crud/booking_utils.py

```diff
@@ -0,0 +1,79 @@
+"""CRUD functions for Booking."""
+
+import datetime
+
+from fastapi import HTTPException
+from sqlalchemy.orm import Session
+
+from crud import client_utils
+from models.booking import Booking
+from schemas.booking_schemas import BookingCreate
+
+
+def get_bookings(db: Session, skip: int = 0, limit: int = 100):
+    """Get all bookings."""
+    return db.query(Booking).offset(skip).limit(limit).all()
+
+
+def get_booking(db: Session, booking_id: int):
+    """Get booking by id."""
+    _booking = db.query(Booking).filter(Booking.id == booking_id).first()
+    if not _booking:
+        raise HTTPException(
+            status_code=404, detail=f"No booking found with id {booking_id}"
+        )
+    return _booking
+
+
+def create_booking(db: Session, booking: BookingCreate):
+    """Create new booking."""
+    if not client_utils.get_client:
+        raise HTTPException(
+            status_code=404,
+            detail=f"No client found with id {booking.client_id}",
+        )
+    if booking.start_date < datetime.date or booking.end_date < datetime.date:
+        raise HTTPException(status_code=400, detail="Incorrect date")
+    _booking = Booking(
+        client_id=booking.client_id,
+        start_date=booking.start_date,
+        end_date=booking.end_date,
+        discount_percent=booking.discount_percent,
+        total_price=booking.total_price,
+        ts_created=datetime.datetime.now,
+        ts_updated=datetime.datetime.now,
+    )
+    db.add(_booking)
+    db.commit()
+    db.refresh(_booking)
+    return _booking
+
+
+def update_booking(db: Session, booking_id: int, booking: BookingCreate):
```

Comment:
booking should be `BookingUpdate`?
and you should update only fields provided in update request

Reply:
Sure, it has to be `BookingUpdate`. Forgot to change the logic for updating bookings and invoices. Will fix that now.